### PR TITLE
Pyro and cryo beams (the genetics powers) adjust temperature more severely, additional effects to make them more meaningfully interact with the world

### PR DESCRIPTION
**Pull request by necromanceranne**
## About The Pull Request

Pyro and cryo beams now adjust temperature by a much larger margin. This is to help them remain useful against....humans. Just straight up naked humans.

Pyro beams can ignite objects and heat reagents within an object. Cryo beams freeze reagents in an object.

Cryo beams directly inflict the freezing blast status effect so that you absolutely will slow someone down on hit. No more fucking around with temp to MAYBE get some kind of effect. 

## Why It's Good For The Game

There was a point where cryokinesis was actually fairly good at slowing people down. Unfortunately, the same person who last touched temperature code also made it nearly unusable against regular human targets due to the rapid pace at which humans normalize their temperature. It takes a good while before temperature slowdown starts to take effect. 

Pyro beams have additional on-hit effects, which cryobeam does not. Cryobeams only have the temperature adjustment and as noted, it is quite weak. Being on fire starts to damage equipment and present a more tangible danger in some environments, such as plasma floods or having just been hit with an accelerant such as alcohol.

By including a status effect onto the cryo beam that forces the target to slow down, we avoid the issue of people ignoring the effect of the cryobeam entirely as a meaningful attack compared to being shot by a pryro beam, which will usually cause people to start panicking or moving to remove the fire in some way.

I also think by adding some additional interactions to the beam to make them meaningfully affect some world objects, like flammable objects, helps to realizes the idea a bit better of them being blasters of temperature rather than just being bullets. It means it goes a little beyond just being a weapon and potentially now a tool. (It was thinking of Bioshock's plasmid advertising when I made this, like lighting cigarettes.)

## Changelog
:cl:
balance: Cryokinesis and pyrokinesis more severely adjust temperature.
balance: Cryokinesis forces a target to slow down on hit for a few seconds.
balance: Pyrokinesis can ignite objects.
balance: Temperature projectiles change the temperature of the target's contained reagents.
/:cl:

### CODE REVIEW

**Comment by moocowswag** [comment]
I recommend locking pyro beam ignite behind power chromosome.

**Comment by necromanceranne** [comment]
I don't really care to do that.

**Comment by Higgin** [comment]
does this at all affect the temp gun projectiles, or do i misread?

**Comment by necromanceranne** [comment]
Nope, separate projectiles.

**Comment by Dawnseer** [comment]
Does this mean I can finally light my cigars/cigarettes with the pyrobeam?

**Comment by Dawnseer** [comment]
Would it be out of scope to instead of being purely an ability button, but instead, turning it into a lil hand item like the shock hand, so that you might use it to interact with say, a cigarette or a molotov? Or as god intended, give flamin high fives?

File: code/modules/projectiles/projectile/special/temperature.dm [hunk 1]
```diff
@@ -32,7 +32,21 @@
...
+
+/obj/projectile/temp/cryo/on_hit(atom/target, blocked = 0, pierce_hit)
+	. = ..()
+
+	if(isobj(target))
+		var/obj/objectification = target
+
+		if(objectification.reagents)
+			var/datum/reagents/reagents = objectification.reagents
+			reagents?.expose_temperature(temperature)
```
Comment (Ghommie) [inline code/modules/projectiles/projectile/special/temperature.dm:45]: I don't quite see why standard temperature projectiles shouldn't do this too.

Comment (necromanceranne) [reply code/modules/projectiles/projectile/special/temperature.dm:45]: *shrug* if you'd like this to be on those, that's entirely doable.

Comment (Ghommie) [reply code/modules/projectiles/projectile/special/temperature.dm:45]: then dew it.

Comment (necromanceranne) [reply code/modules/projectiles/projectile/special/temperature.dm]: I did that

**Pull request merged**